app/controllers/login_controller.rb

```diff
@@ -0,0 +1,37 @@
+class LoginController < ApplicationController
+
+  def index
+  end
+
+  def doauth
+    if session[:authorized] != true
+      authenticate
+      authorize
+    else
+     destroy
+    end
+  end
+
+  def authenticate
+    unless github_authenticated?
+      github_authenticate!
+    end
+  end
+
+  def authorize
+    session[:authorized] ||= github_user.organization_member?('OregonDigital')
+    if session[:authorized] != true
+      flash.keep[:notice] = "authorization failed"
```

Comment:
ja, I can change that. I was probably still flailing around with the flash messages at that point.